src/main/query_profiler.cpp

```diff
@@ -269,11 +294,21 @@ void OperatorProfiler::AddTiming(const PhysicalOperator &op, double time, idx_t
 	auto entry = timings.find(op);
 	if (entry == timings.end()) {
 		// add new entry
-		timings[op] = OperatorInformation(time, elements);
+		timings[op] = OperatorInformation();
+		if (operator_timing_enabled) {
+			timings[op].time = time;
+		}
+		if (operator_cardinality_enabled) {
+			timings[op].elements = elements;
+		}
 	} else {
 		// add to existing entry
-		entry->second.time += time;
-		entry->second.elements += elements;
+		if (operator_timing_enabled) {
+			entry->second.time += time;
+		}
+		if (operator_cardinality_enabled) {
+			entry->second.elements += elements;
+		}
 	}
```

Comment:
Basically because of the `OperatorProfiler` I'm having a hard time figuring out how to add these values to the map, unless I pass the whole map and either store a reference in the struct or pass it around the different functions. What do you think?

Reply:
From my understanding, the `OperatorProfiler` is gathering the profiling information for each operator, i.e., each node. See my comment in `query_profiler.hpp`. Storing a map in each `OperatorProfiler` to set the values to the different keys is fine.